### PR TITLE
[17.0][mig] stock_picking_analytic

### PR DESCRIPTION
**Pull request by Wodran14**

### CODE REVIEW

File: stock_picking_analytic/views/stock_picking.xml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+    <record model="ir.ui.view" id="stock_picking_search_view">
+        <field name="name">stock.picking.search (in stock_picking_analytic)</field>
+        <field name="model">stock.picking</field>
+        <field name="inherit_id" ref="stock.view_picking_internal_search" />
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='name']" position="after">
+                <field
+                    name="analytic_distribution"
+                    readonly="state in ('done', 'cancel')"
```
Comment (astirpe) [inline stock_picking_analytic/views/stock_picking.xml]: This line should be removed:

```suggestion
```

File: stock_picking_analytic/views/stock_picking.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            </xpath>
+            <xpath
+                expr="//notebook/page[@name='detailed_operations']
+                /field[@name='move_line_ids_without_package']"
+                position="attributes"
+            >
+                <attribute name="context" operation="update">
+                    {"default_analytic_distribution": analytic_distribution}
+                </attribute>
+            </xpath>
```
Comment (astirpe) [inline stock_picking_analytic/views/stock_picking.xml]: This part gives error when installing the module. Should it be removed?

```suggestion
```

**Review by astirpe — CHANGES_REQUESTED** [review]

**Comment by peluko00** [comment]
Any update @Wodran14 ?

**Comment by dcorio** [comment]
It's just a warning, but I get:

`warnings.warn(f'Since Odoo 17, property {self}.states is no longer supported.')`

Does it have something to do with this?
```
analytic_distribution = fields.Json(
        inverse="_inverse_analytic_distribution",
        states={
            "done": [("readonly", True)],
            "cancel": [("readonly", True)],
        },
    )
```

**Comment by peluko00** [comment]
Hi @astirpe @dcorio, can you revise this in this another [pr](https://github.com/OCA/account-analytic/pull/676).
Thank you!

**Comment by dreispt** [comment]
Superseded by #628

**Pull request closed without merging**